python/rikai/spark/functions/geometry.py

```diff
@@ -0,0 +1,98 @@
+#  Copyright 2021 Rikai Authors
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""Geometry related PySpark UDF"""
+
+# Third Party
+from pyspark.sql.functions import udf
+from pyspark.sql.types import FloatType
+from rikai.spark.types.geometry import Box2dType
+
+# Rikai
+from rikai.logging import logger
+from rikai.types.geometry import Box2d
+
+__all__ = ["area", "box2d", "box2d_from_center", "box2d_from_top_left"]
+
+
+@udf(returnType=Box2dType())
+def box2d(coords) -> Box2d:
```

Comment:
So the user workflow you're imagining is like: 
1. the bounding boxes will come out of the training model as an numpy array
2. they store that in a column or a struct field
3. then they call this box2d udf on it to convert to Box2d?

Reply:
Right, and for many open dataset (i.e., coco), bounding box are stored as list of numbers in json format.
Here is offering a UDF for conversion.

Reply:
sg that makes sense